pkg/route/validation/validation.go

```diff
@@ -141,9 +180,14 @@ func validateRoute(route *routev1.Route, checkHostname bool) field.ErrorList {
 }
 
 func ValidateRouteUpdate(route *routev1.Route, older *routev1.Route) field.ErrorList {
+	var headersUpdated bool
 	allErrs := validateObjectMetaUpdate(&route.ObjectMeta, &older.ObjectMeta, field.NewPath("metadata"))
 	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(route.Spec.WildcardPolicy, older.Spec.WildcardPolicy, field.NewPath("spec", "wildcardPolicy"))...)
 	hostnameUpdated := route.Spec.Host != older.Spec.Host
+	if route.Spec.HTTPHeaders.Actions.Response != nil || route.Spec.HTTPHeaders.Actions.Request != nil {
+		headersUpdated = cmp.Equal(route.Spec.HTTPHeaders, older.Spec.HTTPHeaders)
+		allErrs = append(allErrs, validateRoute(route, headersUpdated && hostnameUpdated && validLabels(older.Spec.Host))...)
```

Comment:
@candita, I added a if else condition now so that the duplication does not occur. WDYT ?

Reply:
Actually, on second though you should delete all of these changes. The only reason to set the second variable to true is if we need to validate the hostname on a version update.  There were versions where we treated hostname validation differently, and we couldn't reject an already accepted hostname.  It doesn't matter if the header actions changed or not on a version update, because this is a new feature and the validation is new.

Reply:
Please address this, @miheer